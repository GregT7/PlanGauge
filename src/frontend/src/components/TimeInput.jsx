import { TableCell } from "@/components/ui/table";

const TimeInput = ({ task = { id: Date.now(), time_estimation: 0 }, updateTaskField = () => {}, className }) => {
  return (
    <TableCell className={className}>
      <input
        type="number"
        value={task.time_estimation === 0 ? "" : task.time_estimation}
        onChange={(e) => {
          const newValue = Number(e.target.value);
          updateTaskField(task.id, 'time_estimation', newValue);
        }}
        onWheel={(event) => event.currentTarget.blur()}
        className="text-left px-2 w-full h-full bg-transparent outline-none [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none"
      />
    </TableCell>
  );
};


export default TimeInput;