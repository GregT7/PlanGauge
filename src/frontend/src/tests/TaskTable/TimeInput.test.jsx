import { expect, it, describe } from 'vitest';
import userEvent from '@testing-library/user-event';
import { screen, render } from '@testing-library/react';
import TimeInput from "@/components/TaskTable/TimeInput";
import { useState } from 'react';

const TestWrapper = ({ defaultTask }) => {
  const [task, setTask] = useState(defaultTask);

  const updateTaskField = (id, field, value) => {
    setTask((prev) => ({
      ...prev,
      [field]: value,
    }));
  };

  return <TimeInput task={task} updateTaskField={updateTaskField} />;
};


// const TimeInput = ({ task = { id: Date.now(), time_estimation: 0 }, updateTaskField = () => {}, className }) => {
describe("TimeInput unit testing", () => {
    it("passes snapshot test", () => {
        const task = {id: 1, time_estimation: 10}
        const { container } = render(<TimeInput task={task} onChange={() => {}}/>);
        expect(container).toMatchSnapshot();
    });

    it("renders when no props are passed", () => {
        const { container } = render(<TimeInput/>);
        expect(container).toBeInTheDocument();
    });

    it ("updates state when user enters an integer", async () => {
        const task = {id: 1, time_estimation: 10}
        render(<TestWrapper defaultTask={task}/>)

        const user = userEvent.setup();
        const numInput = screen.getByRole('spinbutton')
        await user.type(numInput, '567');
        expect(numInput).toHaveValue(10567);
    });

    it ("does not update state when user enters a non-integer", async () => {
        const task = {id: 1, time_estimation: 10}
        render(<TestWrapper defaultTask={task}/>)

        const user = userEvent.setup();
        const numInput = screen.getByRole('spinbutton')
        await user.type(numInput, 'aa123aabc456');
        expect(numInput).toHaveValue(10123456);
    });
});