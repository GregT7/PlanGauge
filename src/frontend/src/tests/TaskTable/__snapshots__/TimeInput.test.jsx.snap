// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TimeInput unit testing > passes snapshot test 1`] = `
<div>
  <td
    class="h-10 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
    data-slot="table-cell"
  >
    <input
      class="text-left px-2 w-full h-full bg-transparent outline-none [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none"
      type="number"
      value="10"
    />
  </td>
</div>
`;
